evm/test/TrimmedAmount.t.sol

```diff
@@ -156,4 +156,112 @@ contract TrimmingTest is Test {
 
         assertEq(expectedRoundTrip, amountRoundTrip);
     }
+
+    // invariant: forall (x: uint256, y: uint8, z: uint8),
+    //            (x <= type(uint64).max, y <= z)
+    //                    => (x.trim(x, 8).untrim(y) == x)
+    function testFuzz_trimIsLeftInverse(
+        uint256 amount,
+        uint8 fromDecimals,
+        uint8 toDecimals
+    ) public {
+        // this is guaranteed by trimming
+        uint256 amt = bound(amount, 1, type(uint64).max);
+        vm.assume(fromDecimals <= 18);
+        vm.assume(toDecimals <= 18);
+
+        // trimming
+        uint8 initialDecimals = 0;
+        TrimmedAmount memory trimmedAmount = TrimmedAmount(uint64(amt), initialDecimals);
+
+        // trimming is left inverse of trimming
+        uint256 amountUntrimmed = trimmedAmount.untrim(fromDecimals);
+        TrimmedAmount memory amountRoundTrip = amountUntrimmed.trim(fromDecimals, initialDecimals);
+
+        assertEq(trimmedAmount.amount, amountRoundTrip.amount);
+    }
+
+    // FUZZ TESTS
+
+    // invariant: forall (TrimmedAmount a, TrimmedAmount b) a.saturatingAdd(b).amount <= type(uint64).max
+    function testFuzz_saturatingAddDoesNotOverflow(
+        TrimmedAmount memory a,
+        TrimmedAmount memory b
+    ) public {
+        vm.assume(a.decimals == b.decimals);
+
+        // this is guaranteed by trimming
+        a.amount = uint64(bound(a.amount, 0, type(uint64).max));
+        a.decimals = uint8(bound(a.decimals, 0, 8));
+
+        b.amount = uint64(bound(b.amount, 0, type(uint64).max));
+        b.decimals = uint8(bound(b.decimals, 0, 8));
```

Comment:
Why only bounding to 8 decimals here and don't think there's any need to bound the amounts since it's already a uint64?